uiowa/uiowa
# [Facilities] Update Dispatch to use custom fields in subscriber signup

# How to test



```
ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local uli admin/config/sitenow/sitenow-dispatch
```
Go to [facilities.uiowa.edu](https://www.facilities.uiowa.edu/admin/config/services/facilities-alerts-emails) and get the dispatch API key and enter it here https://facilities.uiowa.ddev.site/admin/config/sitenow/sitenow-dispatch

Edit the Dispatch base path locally to pull from the test endpoint: https://test.its.uiowa.edu/dispatch/api/v1/
https://github.com/uiowa/uiowa/blob/4918096632bfc29d74dcf7d6e3ec562023a97102/docroot/modules/custom/sitenow_dispatch/src/DispatchApiClient.php#L44-L46

Check out the Alerts sign up form at https://facilities.uiowa.ddev.site/alerts

Configure the signup form in the [Dispatch test environment](https://test.its.uiowa.edu/dispatch/populations/574308527/configure) and add new fields and toggle the display of the phone number field

See that the changes are reflected in the SiteNow form.

Turn on xdebug and submit the form, catching it prior to https://github.com/uiowa/uiowa/blob/4918096632bfc29d74dcf7d6e3ec562023a97102/docroot/modules/custom/sitenow_dispatch/src/Form/SubscribeForm.php#L93C1-L99C8

Check that each parameter matches the keys for the Dispatch side of things, and with strings containing the values entered into the SiteNow form.

## Followup

- [x] Communicate with Michael A if this all looks good for things to go into prod on their end
